comp/serializer/compression/compressionimpl/zlib_and_zstd.go

```diff
@@ -0,0 +1,43 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build zlib && zstd
```

Comment:
do we need this when we are reading from config ?

Reply:
Yeah because there is a version of this without zstd. And in that case, the agent still respects the config but will not supply a zstd implementation. So we need to build tag them to preserve both build time compression options and runtime compression settings